src/renderer/components/cluster-manager/cluster-status.tsx

```diff
@@ -153,7 +153,7 @@ export class ClusterStatus extends React.Component<Props> {
     return (
       <div className={cssNames(styles.status, "flex column box center align-center justify-center", this.props.className)}>
         <div className="flex items-center column gaps">
-          <h2>{this.entity.getName()}</h2>
+          <h2>{this.entity?.getName()}</h2>
```

Comment:
It's a race condition or why we wouldn't have `entity` on this page?

Reply:
Yes I think it is a race condition. I have only seen this issue when I reload the top view.

Reply:
This will possibly set the name to `undefined`? Seems fishy that `entity` can be undefined. What was wrong with `this.cluster.preferences.clusterName`? At least as a fallback?

Reply:
That is a good idea as a fallback.